core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -3344,50 +3344,109 @@ public String toString() {
   // Job service
   //
   public static final PropertyKey JOB_MASTER_CLIENT_THREADS =
-      new Builder(Name.JOB_MASTER_CLIENT_THREADS).setDefaultValue(1024).build();
-  public static final PropertyKey JOB_MASTER_FINISHED_JOB_RETENTION_MS =
-      new Builder(Name.JOB_MASTER_FINISHED_JOB_RETENTION_MS).setDefaultValue(300000).build();
+      new Builder(Name.JOB_MASTER_CLIENT_THREADS)
+          .setDescription("The number of threads the Alluxio master uses to make requests to the "
+              + "job master.")
+          .setDefaultValue(1024)
+          .build();
+  public static final PropertyKey JOB_MASTER_FINISHED_JOB_RETENTION_TIME =
+      new Builder(Name.JOB_MASTER_FINISHED_JOB_RETENTION_TIME)
+          .setDescription("The length of time the Alluxio Job Master should save information about "
+              + "completed jobs before they are discarded.")
+          .setDefaultValue("300sec")
+          .build();
   public static final PropertyKey JOB_MASTER_JOB_CAPACITY =
-      new Builder(Name.JOB_MASTER_JOB_CAPACITY).setDefaultValue(100000).build();
-  public static final PropertyKey JOB_MASTER_WORKER_HEARTBEAT_INTERVAL_MS =
-      new Builder(Name.JOB_MASTER_WORKER_HEARTBEAT_INTERVAL_MS).setDefaultValue(1000).build();
-  public static final PropertyKey JOB_MASTER_WORKER_TIMEOUT_MS =
-      new Builder(Name.JOB_MASTER_WORKER_TIMEOUT_MS).setDefaultValue(60000).build();
+      new Builder(Name.JOB_MASTER_JOB_CAPACITY)
+          .setDescription("The total number of jobs that the ALluxio Job Master can run at any one "
```

Comment:
After taking a look at the code, it looks like finished jobs will count towards this number if they are within the retention period. As far as I can tell if the queue is full due to jobs, but they are within retention policy, it will count towards the number here. https://github.com/Alluxio/alluxio/blob/master/job/server/src/main/java/alluxio/master/job/JobMaster.java#L202-L228

Reply:
Do you think this would be more helpful?

"The total number of jobs that the Alluxio Job Master can run at any one moment. Jobs which have finished within `alluxio.job.master.finished.job.retention.time` will count towards this value."

Reply:
So it's basically a bound on `num_running_jobs` + `num_retained_results`

Reply:
I think the main use case for this property is to control how much master memory is consumed by job results. The limitation on `num_running_jobs` is just a side effect